v2/component/launcher.go

```diff
@@ -0,0 +1,438 @@
+
+		b, err := ioutil.ReadFile(v.FileInputPath)
+		if err != nil {
+			return fmt.Errorf("Failed to read input artifact metadata file for %q: %v", k, err)
+		}
+
+		a := &pb.Artifact{}
+		if err := protojson.Unmarshal(b, a); err != nil {
+			return fmt.Errorf("Failed to unmarshall input artifact metadata for %q: %v", k, err)
+		}
+
+		v.Artifact = a
+
+		// Prepare input uri placeholder.
+		key := fmt.Sprintf(`{{$.inputs.artifacts['%s'].uri}}`, k)
+		l.placeholderReplacements[key] = v.Artifact.GetUri()
+
+		// Prepare input path placeholder.
+		v.LocalArtifactFilePath = path.Join("/tmp/kfp_launcher_inputs", k, "data")
+		key = fmt.Sprintf(`{{$.inputs.artifacts['%s'].path}}`, k)
+		l.placeholderReplacements[key] = v.LocalArtifactFilePath
+
+		// Copy artifact to local storage.
+		// TODO: Selectively copy artifacts for which .path was actually specified
+		// on the command line.
+
+		blobKey, err := l.bucketConfig.keyFromURI(v.Artifact.GetUri())
+		if err != nil {
+			return err
+		}
+		r, err := bucket.NewReader(ctx, blobKey, nil)
+		if err != nil {
+			return err
+		}
+		defer r.Close()
+
+		if err := os.MkdirAll(path.Dir(v.LocalArtifactFilePath), 0644); err != nil {
+			return err
+		}
+		w, err := os.Create(v.LocalArtifactFilePath)
+		if err != nil {
+			return err
+		}
+		if _, err := io.Copy(w, r); err != nil {
+			return err
+		}
+	}
+
+	// Prepare input parameter placeholders.
+	for k, v := range l.runtimeInfo.InputParameters {
+		key := fmt.Sprintf(`{{$.inputs.parameters['%s']}}`, k)
+		l.placeholderReplacements[key] = v.ParameterValue
+	}
+
+	return nil
+}
+
+func (l *Launcher) prepareOutputs(ctx context.Context) error {
+	for k, v := range l.runtimeInfo.OutputParameters {
+		key := fmt.Sprintf(`{{$.outputs.parameters['%s'].output_file}}`, k)
+		l.placeholderReplacements[key] = v.FileOutputPath
+	}
+
+	for k, v := range l.runtimeInfo.OutputArtifacts {
+		// TODO: sanitize k
+		v.LocalArtifactFilePath = path.Join("/tmp/kfp_launcher_outputs", k, "data")
+
+		if err := os.MkdirAll(path.Dir(v.LocalArtifactFilePath), 0644); err != nil {
+			return err
+		}
+
+		blobKey := path.Join(l.options.PipelineName, l.options.PipelineRunID, l.options.PipelineTaskID, "data")
+		v.URIOutputPath = l.bucketConfig.uriFromKey(blobKey)
+
+		key := fmt.Sprintf(`{{$.outputs.artifacts['%s'].path}}`, k)
+		l.placeholderReplacements[key] = v.LocalArtifactFilePath
+
+		key = fmt.Sprintf(`{{$.outputs.artifacts['%s'].uri}}`, k)
+		l.placeholderReplacements[key] = v.URIOutputPath
+	}
+
+	return nil
+}
+
+// RunComponent runs the current KFP component using the specified command and
+// arguments.
+func (l *Launcher) RunComponent(ctx context.Context, cmd string, args ...string) error {
+
+	if err := l.prepareInputs(ctx); err != nil {
+		return err
+	}
+
+	if err := l.prepareOutputs(ctx); err != nil {
+		return err
+	}
+
+	// Update command.
+	for i, v := range args {
+		if _, ok := l.placeholderReplacements[v]; ok {
+			args[i] = l.placeholderReplacements[v]
+		}
+	}
+
```

Comment:
Shall we throw an err incase there's any unresolved placeholder left in the args?